torch/csrc/jit/python/python_ivalue.h

```diff
@@ -26,7 +26,8 @@ struct C10_EXPORT ConcretePyObjectHolder final : PyObjectHolder {
 
   ~ConcretePyObjectHolder() {
     pybind11::gil_scoped_acquire ag;
-    py_obj_ = py::none();
+    py_obj_.dec_ref();
+    py_obj_.ptr() = nullptr;
   }
```

Comment:
How about adding detailed explanation as well.

2 points
- Why `py_obj_ = py::none();` does not work. Because we also need to acquire GIL when destructing py::object of None which de-references None. https://docs.python.org/3/c-api/none.html#c.Py_RETURN_NONE
- Why we need to call `dec_ref()` explicitly. Because py::object of nullptr, on destruction, effectively does nothing because of it calls `Py_XDECREF(NULL)` underlying. https://docs.python.org/3/c-api/refcounting.html#c.Py_XDECREF

Reply:
Thanks for writing this down. Will add this to the PR.